### PR TITLE
Feature: setPeriodic add argument initialDelay parameter

### PR DESCRIPTION
**Pull request by pigbayspy**
Motivation:

add a new parameter `initialDelay` in `setPeriodic` method of `Vertx` https://github.com/eclipse-vertx/vert.x/issues/4436

### CODE REVIEW

**Review by tsegismont — COMMENTED** [review]
@pigbayspy thank you! Can you please also make the changes for `periodicStream`?

**Comment by pigbayspy** [comment]
> @pigbayspy thank you! Can you please also make the changes for `periodicStream`?

No problem. I would do this later

**Comment by pigbayspy** [comment]
I've done it @tsegismont

File: src/main/java/io/vertx/core/impl/VertxImpl.java [hunk 1]
```diff
@@ -965,6 +992,14 @@ private class TimeoutStreamImpl implements TimeoutStream, Handler<Long> {
     private long demand;
 
     public TimeoutStreamImpl(long delay, boolean periodic) {
+      this.initialDelay = delay;
+      this.delay = delay;
+      this.periodic = periodic;
+      this.demand = Long.MAX_VALUE;
```
Comment (tsegismont) [inline src/main/java/io/vertx/core/impl/VertxImpl.java:998]:
```suggestion
      this(delay, delay, periodic);
```

File: src/main/java/io/vertx/core/impl/VertxImpl.java [hunk 2]
```diff
@@ -357,9 +357,22 @@ public EventBus eventBus() {
     return eventBus;
   }
 
+  @Override
   public long setPeriodic(long delay, Handler<Long> handler) {
     ContextInternal ctx = getOrCreateContext();
-    return scheduleTimeout(ctx, true, delay, TimeUnit.MILLISECONDS, ctx.isDeployment(), handler);
+    return scheduleTimeout(ctx, true, delay, delay, TimeUnit.MILLISECONDS, ctx.isDeployment(), handler);
+  }
```
Comment (tsegismont) [inline src/main/java/io/vertx/core/impl/VertxImpl.java:364]: Here you can delegate to the other method `setPeriodic` method.

**Review by tsegismont — COMMENTED** [review]
Thank you. Just a few more suggestions

**Review by tsegismont — APPROVED** [review]
Thank you @pigbayspy !

**Comment by tsegismont** [comment]
@vietj I'll wait for the 4.3.3 release to be complete before merging this

**Comment by tsegismont** [comment]
Windows build failure looks unrelated

**Pull request merged**